build failure w/sanitycheck for samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4

On multiple platform seeing failures if we do:

`./scripts/sanitycheck -s samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 -l`

we get:
```
ERROR   - bl654_dvk                 samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/bl654_dvk/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    1/  16   6%  skipped:    0, failed:    1
ERROR   - nrf52840dk_nrf52840       samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/nrf52840dk_nrf52840/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    2/  16  12%  skipped:    0, failed:    2
ERROR   - bl653_dvk                 samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/bl653_dvk/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    3/  16  18%  skipped:    0, failed:    3
ERROR   - particle_boron            samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/particle_boron/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    4/  16  25%  skipped:    0, failed:    4
ERROR   - reel_board_v2             samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/reel_board_v2/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    5/  16  31%  skipped:    0, failed:    5
ERROR   - nrf52833dk_nrf52833       samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/nrf52833dk_nrf52833/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    6/  16  37%  skipped:    0, failed:    6
ERROR   - pinnacle_100_dvk          samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/pinnacle_100_dvk/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    7/  16  43%  skipped:    0, failed:    7
ERROR   - nrf52840dongle_nrf52840   samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/nrf52840dongle_nrf52840/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    8/  16  50%  skipped:    0, failed:    8
ERROR   - particle_xenon            samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/particle_xenon/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:    9/  16  56%  skipped:    0, failed:    9
ERROR   - nrf52840_blip             samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/nrf52840_blip/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:   10/  16  62%  skipped:    0, failed:   10
ERROR   - reel_board                samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/reel_board/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:   11/  16  68%  skipped:    0, failed:   11
ERROR   - particle_argon            samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/particle_argon/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:   12/  16  75%  skipped:    0, failed:   12
ERROR   - nrf52840_papyr            samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/nrf52840_papyr/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:   13/  16  81%  skipped:    0, failed:   13
ERROR   - nrf52840_mdk              samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/nrf52840_mdk/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:   14/  16  87%  skipped:    0, failed:   14
ERROR   - adafruit_feather_nrf52840 samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/adafruit_feather_nrf52840/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log
INFO    - Total complete:   15/  16  93%  skipped:    0, failed:   15
ERROR   - nrf52833dk_nrf52820       samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4 FAILED: Build failure
ERROR   - see: /home/galak/git/zephyr/sanity-out/nrf52833dk_nrf52820/samples/bluetooth/hci_usb_h4/sample.bluetooth.hci_usb_h4/build.log

```

From the log:
```
/home/galak/git/zephyr/subsys/bluetooth/controller/ll_sw/ull_scan.c: In function 'll_scan_enable':
/home/galak/git/zephyr/subsys/bluetooth/controller/ll_sw/ull_scan.c:130:10: error: unused variable 'is_update_1m' [-Werror=unused-variable]
  130 |  uint8_t is_update_1m = 0U;
      |          ^~~~~~~~~~~~
/home/galak/git/zephyr/subsys/bluetooth/controller/ll_sw/ull_scan.c:127:10: error: unused variable 'is_update_coded' [-Werror=unused-variable]
  127 |  uint8_t is_update_coded = 0U;
      |          ^~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/bluetooth/controller/ll_sw/ull_scan.c:125:22: error: unused variable 'node_rx_scan_term' [-Werror=unused-variable]
  125 |  struct node_rx_pdu *node_rx_scan_term = NULL;
      |                      ^~~~~~~~~~~~~~~~~

```

--- a/subsys/bluetooth/controller/ll_sw/ull_scan.c
+++ b/subsys/bluetooth/controller/ll_sw/ull_scan.c
@@ -118,16 +118,17 @@ uint8_t ll_scan_params_set(uint8_t type, uint16_t interval, uint16_t window,
 
 #if defined(CONFIG_BT_CTLR_ADV_EXT)
 uint8_t ll_scan_enable(uint8_t enable, uint16_t duration, uint16_t period)
-#else /* !CONFIG_BT_CTLR_ADV_EXT */
-uint8_t ll_scan_enable(uint8_t enable)
-#endif /* !CONFIG_BT_CTLR_ADV_EXT */
 {
 	struct node_rx_pdu *node_rx_scan_term = NULL;
-	struct ll_scan_set *scan_coded = NULL;
 	uint8_t is_update_coded = 0U;
+	uint8_t is_update_1m = 0U;
+#else /* !CONFIG_BT_CTLR_ADV_EXT */
+uint8_t ll_scan_enable(uint8_t enable)
+{
+#endif /* !CONFIG_BT_CTLR_ADV_EXT */
+	struct ll_scan_set *scan_coded = NULL;
 	uint8_t own_addr_type = 0U;
 	uint8_t is_coded_phy = 0U;
-	uint8_t is_update_1m = 0U;
 	struct ll_scan_set *scan;
 	uint8_t err;
 

--- a/subsys/bluetooth/controller/ll_sw/ull_scan.c
+++ b/subsys/bluetooth/controller/ll_sw/ull_scan.c
@@ -163,13 +163,10 @@ uint8_t ll_scan_enable(uint8_t enable)
 #if defined(CONFIG_BT_CTLR_ADV_EXT) && defined(CONFIG_BT_CTLR_PHY_CODED)
 	scan_coded = ull_scan_is_disabled_get(SCAN_HANDLE_PHY_CODED);
 	if (!scan_coded) {
-#if defined(CONFIG_BT_CTLR_ADV_EXT)
 		is_update_coded = is_scan_update(SCAN_HANDLE_PHY_CODED,
 						 duration, period, &scan_coded,
 						 &node_rx_scan_term);
-		if (!is_update_coded)
-#endif /* CONFIG_BT_CTLR_ADV_EXT */
-		{
+		if (!is_update_coded) {
 			return BT_HCI_ERR_CMD_DISALLOWED;
 		}
 	}

